x-pack/plugins/apm/common/environment_filter_values.ts

```diff
@@ -35,6 +37,22 @@ export function getEnvironmentLabel(environment: string) {
   return environmentLabels[environment] || environment;
 }
 
+export function parseEnvironmentUrlParam(environment: string) {
+  if (environment === ENVIRONMENT_ALL_VALUE) {
+    return ENVIRONMENT_ALL;
+  }
+
+  if (environment === ENVIRONMENT_NOT_DEFINED_VALUE) {
+    return ENVIRONMENT_NOT_DEFINED;
+  }
+
+  return {
+    esFieldValue: environment,
+    value: environment,
+    text: environment,
+  };
+}
```

Comment:
Afaict this should be:

```ts
export function getEsFieldValue(environment: string) {
    if (environment === ENVIRONMENT_ALL_VALUE) {
      return ENVIRONMENT_ALL;
    }
  
    if (environment === ENVIRONMENT_NOT_DEFINED_VALUE) {
      return ENVIRONMENT_NOT_DEFINED;
    }
  
    return environment
}
```

Then you can call it the specific places where needed (currently just for alerts)